examples/step-13/step-13.cc

```diff
@@ -737,22 +737,22 @@ namespace Step13
       Threads::Task<> rhs_task =
         Threads::new_task(&Solver<dim>::assemble_rhs, *this, linear_system.rhs);
 
-      WorkStream::run(dof_handler.begin_active(),
-                      dof_handler.end(),
-                      std::bind(&Solver<dim>::local_assemble_matrix,
-                                this,
-                                std::placeholders::_1,
-                                std::placeholders::_2,
-                                std::placeholders::_3),
-                      std::bind(&Solver<dim>::copy_local_to_global,
-                                this,
-                                std::placeholders::_1,
-                                std::ref(linear_system)),
-                      AssemblyScratchData(*fe, *quadrature),
-                      AssemblyCopyData());
+      WorkStream::run(
+        dof_handler.begin_active(),
+        dof_handler.end(),
+        [this](const typename DoFHandler<dim>::active_cell_iterator &cell,
+               AssemblyScratchData &scratch_data,
+               AssemblyCopyData &   copy_data) {
+          this->local_assemble_matrix(cell, scratch_data, copy_data);
+        },
+        [this, &linear_system](const AssemblyCopyData &copy_data) {
```

Comment:
Since this is in a tutorial program where simplicity is a concern, what would you think about omitting the explicit captures here?

Reply:
I don't feel that listing the captures is a big issue, but I find it a lot more readable if you move the definition of the lambda above the function call and give it a descriptive name.

Reply:
My reasoning for the explicit capture list would be that we make it as clear as possible what the lambda is really doing (and which variables it is using). 
I don't have a strong opinion here, though. If someone strongly prefers one variant I am happy to change all of them in the examples.

I think that we should discuss both variants here in any way.

Reply:
Also see http://open-std.org/JTC1/SC22/WG21/docs/papers/2018/p0806r2.html and https://en.cppreference.com/w/cpp/language/lambda:
```
 The implicit capture of *this when the capture default is = is deprecated. (since C++20).
```